IA-2131  / IA-2133: API to return orgunit without submissions

olethanh opened the pull request:
fix IA-2131 API to return orgunit without submissions
fix also IA-2133

this is needed for the Registry page "Missing feature submissions"
After discussions we are implementing that in the completeness api

This add a new argument `without_submissions` that only return the orgunit who have no submission for the queryied `form_id`(s)

Example call /api/v2/completeness_stats/?page=1&limit=50&form_id=6&parent_org_unit_id=730&without_submissions=true
this is paginated and accept the same habitual arguments for the completess_api (See swagger  for doc )

Note that per default the completeness api only return the direct
children, pass the `org_unit_type_ids` if you want all the descendants
of that type(s)



## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [NA] New translations have been added or updated if new strings have been introduced in the frontend
- [NA] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes
see above
## How to test

Find an orgunit whose child should enter a submission but hasn't  (red cross) in the completeness ui, check what network call it's making and add the arguments. or let Christophe implement the front



Adding test later

quang-le reviewed (changes requested):
Only two blocking issues: 
- change `initialUser` from callback to memo
- fix cypress tests (not passing in terminal, nor in Chrome)

Other comments and suggestions are mostly stylistic

The code is really much cleaner and easier to read now

beygorghor commented on hat/assets/js/apps/Iaso/domains/users/hooks/useGetProfiles.js:
```diff
@@ -0,0 +1,18 @@
+import { useSnackQuery } from 'Iaso/libs/apiHooks';
+
+export const useGetProfiles = params => {
+    const nemParams = {
```
We are doing the same thing a bit everywhere...

olethanh committed:
IA-2131 API to return orgunit without submissions

this is needed for the Registry page "Missing feature submissions"
After discussions we are implementing that in the completeness api

This add a new argument `without_submissions` that only return the orgunit who have no submission for the queryied `form_id`(s)

Example call /api/v2/completeness_stats/?page=1&limit=50&form_id=6&parent_org_unit_id=730&without_submissions=true
this is paginated and accept the same habitual arguments for the completess_api (See swagger  for doc )

Note that per default the completeness api only return the direct
children, pass the `org_unit_type_ids` if you want all the descendants
of that type(s)

olethanh committed:
Merge remote-tracking branch 'origin/main' into IA-2131-api-to-return-orgunit-without-submission

olethanh committed:
Add test for without_submissison IA-2131

olethanh committed:
Try to fix casting error

 i think it's due to older postgresql version in the ci

olethanh committed:
Merge remote-tracking branch 'origin/main' into IA-2131-api-to-return-orgunit-without-submission

olethanh committed:
Merge remote-tracking branch 'origin/main' into IA-2131-api-to-return-orgunit-without-submission

olethanh committed:
Casting error correcty fixed this time

beygorghor committed:
IA-2133: fix multiple projects
